mailpile/plugins/__init__.py

```diff
@@ -782,6 +776,15 @@ def get_ui_elements(self, ui_type, context):
         return [elem for elem in self.UI_ELEMENTS[ui_type]
                 if context in elem['context']]
 
+    ##[ Pluggable Hooks ]####################################################
+
+    INCOMING_EMAIL = {}
```

Comment:
I don't like how completely unstructured this is, explicit is better than implicit if we want to avoid weird heisenbugs.

Also note the comments about where this is invoked; I don't think this is integrated at the right place at the moment, and I think the functionality you want already exists (but may have a misleading name).